.github/README.md

```diff
@@ -0,0 +1,95 @@
+|------------------------|-------------------------------------------------------------------------------------------------------------------------------|------|
+| lint                   | Runs linting and ensures vendor directory, protos and generated documentation are consistent.                                 | CI   |
+| test                   | Runs units tests on Cassandra testing framework.                                                                              | CI   |
+| integration-configs-db | Integration tests for database configurations.                                                                                | CI   |
+| integration            | Runs integration tests after upgrading golang, pulling necessary docker images and downloading necessary module dependencies. | CI   |
+| build                  | Builds and saves an up-to-date Cortex image and website.                                                                      | CI   |
+| deploy_website         | Deploys the latest version of Cortex website to gh-pages branch. Triggered within workflow.                                   | CD   |
+| deploy                 | Deploys the latest Cortex image.                                                                                              | CD   |
+
+## Job Dependency Graph
+
+Internal dependencies between jobs illustrated below. Jobs run concurrently where possible but do not start until all jobs they depend on have completed successfully.
+
+
+![cortex_test-build-deploy](https://user-images.githubusercontent.com/20804975/95492784-9b7feb80-0969-11eb-9934-f44a4b1da498.png)
+
+### Key Details
+
+**Naming Convention**
+
+Each step in a job has a clear name that encapsulates the purpose of the command. The convention we are using is each word in the name should be capitalized except articles and prepositions. This creates consistent labeling when looking at the progress of the current workflow on GitHub.
+
+```yaml
+- name: Checkout Repo
+# commands
+- name: Get Dependencies
+# commands
+```
+
+**Checkout Version**
+
+Current build-image ships with an older version of Git which breaks github/actions@v2 so we are using actions/checkout@v1 for all jobs until the quay image is updated to ship with a more recent version of Git.
+
+```yaml
+- name: Checkout Repo
+  uses: actions/checkout@v1
+```
+
+**Symbolic Link to Expected Workspace**
+
+A significant number of commands in the Makefile are hardcoded with an assumed file structure of the CI container. To ensure paths specified in previous commands don’t break, a symlink was created from the hardcoded “expected” working directory `/go/src/github.com/cortexproject/cortex` to the actual working directory `$GITHUB_WORKSPACE`.
+
+```yaml
+- name: Sym link expected path to github workspace
+  run: |
+    mkdir -p /go/src/github.com/cortexproject/cortex
+    ln -s $GITHUB_WORKSPACE/* /go/src/github.com/cortexproject/cortex
```

Comment:
This looks suspicious... why create many links instead of one:

```
mkdir -p /go/src/github.com/cortexproject/
ln -s "$GITHUB_WORKSPACE" /go/src/github.com/cortexproject/cortex
```